distributed/scheduler.py

```diff
@@ -569,7 +569,9 @@ def __hash__(self) -> int:
         return self._hash
 
     def __eq__(self, other: object) -> bool:
-        return isinstance(other, WorkerState) and other.server_id == self.server_id
+        return self is other or (
```

Comment:
This should speed up equality checks since we should mostly (always?) be using the same instance anyways.

Reply:
should always be the same. I'd even be fine with removing `__eq__` entirely in which case `==` is the same as `is`

Reply:
> I'd even be fine with removing `__eq__` entirely in which case `==` is the same as `is`

I like the idea, but I'd leave it for a separate PR.